pkg/cmd/roachtest/tests/knex.go

```
@@ -0,0 +1,130 @@
+
+// This test runs one of knex's test suite against a single cockroach
+// node.
+
+func registerKnex(r registry.Registry) {
+	runKnex := func(
+		ctx context.Context,
+		t test.Test,
+		c cluster.Cluster,
+	) {
+		if c.IsLocal() {
+			t.Fatal("cannot be run in local mode")
+		}
+		node := c.Node(1)
+		t.Status("setting up cockroach")
+		c.Put(ctx, t.Cockroach(), "./cockroach", c.All())
+		if err := c.PutLibraries(ctx, "./lib"); err != nil {
+			t.Fatal(err)
+		}
+		c.Start(ctx, c.All())
+
+		version, err := fetchCockroachVersion(ctx, c, node[0])
+		require.NoError(t, err)
+
+		err = alterZoneConfigAndClusterSettings(ctx, version, c, node[0])
+		require.NoError(t, err)
+
+		err = repeatRunE(
+			ctx,
+			t,
+			c,
+			node,
+			"create sql database",
+			`./cockroach sql --insecure -e "CREATE DATABASE test"`,
+		)
+		require.NoError(t, err)
+
+		err = repeatRunE(
+			ctx,
+			t,
+			c,
+			node,
+			"add nodesource repository",
+			`sudo apt install ca-certificates && curl -sL https://deb.nodesource.com/setup_12.x | sudo -E bash -`,
+		)
+		require.NoError(t, err)
+
+		err = repeatRunE(
+			ctx, t, c, node, "install nodejs and npm", `sudo apt-get -qq install nodejs`,
+		)
+		require.NoError(t, err)
+
+		err = repeatRunE(
+			ctx, t, c, node, "update npm", `sudo npm i -g npm`,
+		)
+		require.NoError(t, err)
+
+		err = repeatRunE(
+			ctx, t, c, node, "install mocha", `sudo npm i -g mocha`,
+		)
+		require.NoError(t, err)
+
+		err = repeatRunE(
+			ctx, t, c, node, "remove old knex", `sudo rm -rf /mnt/data1/knex`,
+		)
+		require.NoError(t, err)
+
+		err = repeatGitCloneE(
+			ctx,
+			t,
+			c,
+			"https://github.com/knex/knex.git",
+			"/mnt/data1/knex",
+			supportedKnexTag,
+			node,
+		)
+		require.NoError(t, err)
+
+		err = repeatRunE(
+			ctx, t, c, node, "install knex npm dependencies", `cd /mnt/data1/knex/ && sudo npm i`,
+		)
+		require.NoError(t, err)
+
+		t.Status("running knex tests")
+		// Ignore the error, this is expected to fail.
```

this says ignore the error, but line 117 calls fatal on it. i think in this case since all  tests pass, the `fatal` is correct, so the comment should be updated